### ISSUE TITLE
When Buy button is clicked, it calls Node.downloadFile() instead of making a retrieval deal

### ISSUE THREAD

**Issue by ghost** [agencyenterprise/filecoin-retrieval-browser-extension#53]
tl;dr When a Buy button is clicked, we need to initiate a proper retrieval deal with the other browser using Client.retrieve().

When a Buy button is clicked:
 -  it looks like it calls Offers.downloadFile() in `src/popup/App/Home/Offers/Offers.js`
 - That function sends a message to the background like this:  `chrome.runtime.sendMessage({ messageType: messageTypes.downloadFile, msg });`
 - messageTypes.downloadFile calls Node.downloadFile(), which uses `chrome.downloads.download()`.  That transfers the file for free.

What we want instead is to use the `class Client` interface in `src/background/retrieval-market/Client.js` to kick off a retrieval deal.  Since Node object already constructs the Client class and saves it as `this.client`, we should just add a Node method [like described here](https://github.com/filecoin-shipyard/browser-retrieval/issues/53#issuecomment-708712703).

**Comment by ghost** [comment]
@brunolm Do you know where the method `Node.downloadFile()` came from?  If it was just implemented to make the Buy button work, let's remove it.  

If it existed before the Buy button, let's leave it for now while we figure out its purpose.

**Comment by brunolm** [comment]
@mgoelzer it was already there. I didn't create it.

Before the "Offers" feature it was used to start downloading the file when you queried a CID. When I implemented the "Offers" I moved the download process to the "Buy" button.

**Comment by ghost** [comment]
Here's what we need to do then.  When the user queries a cid, one or more json strings are going to come in from each potential browser provider.  They contain `{ cid, params, multiaddrs: [multiaddr] }`.  We should associate each Buy button with this object for that particular Buy offer.

Then, when a Buy button is pressed, we should pass those parameters to a method in Node like this:
```
  async initiateRetrieval({ cid, params, multiaddrs: [multiaddr] }) {
    if (this.queriedCids.has(cid)) {
      try {
        this.queriedCids.delete(cid);   // <--- I'm not sure what this does.  Should we keep it or not?
        await this.client.retrieve(cid, params, multiaddr);
      } catch (error) {
        console.error(error);
        ports.postLog(`ERROR: Node.initiateRetrieval():  failed: ${error.message}`);
      }
    }
  }
```
